zerver/lib/actions.py

```diff
@@ -4591,17 +4592,27 @@ def do_update_message(user_profile: UserProfile, message: Message,
         ]
         # Users who can longer access the message without some action
         # from administrators.
-        #
-        # TODO: Extend this list to also contain users losing access
-        # due to the messages moving to a private stream they are not
-        # subscribed to.
-        subs_losing_access = [
-            sub for sub in subs_losing_usermessages
-            if sub.user_profile.is_guest
-        ]
+        if not new_stream.is_public():
+            subs_losing_access = subs_losing_usermessages
+        else:
+            subs_losing_access = [
+                sub for sub in subs_losing_usermessages
+                if sub.user_profile.is_guest
+            ]
```

Comment:
I think this can be more simply expressed like this; I'll go ahead and make the change:

```
subs_losing_access = [
    sub for sub in subs_losing_usermessages
    if sub.user_profile.is_guest or not new_stream.is_public()
]
```

Reply:
Thanks!